### ISSUE TITLE
Deepseek coder v2 is providing gibberish output

### ISSUE THREAD

**Issue by Manik04IISER** [ollama/ollama#5339]
### What is the issue?
The Model being Deepseek Coder v2 16b q: 5_K_M
I provided a code block to the model and it started to produce gibberish. Whereas for any other model, it works fine.
![Screenshot_2024-06-28-00-46-09_1920x1080](https://github.com/ollama/ollama/assets/120251924/fceb7566-cc56-4537-bbb4-6ad431b92db7)

The log file :
![Screenshot_2024-06-28-00-48-23_1920x1080](https://github.com/ollama/ollama/assets/120251924/4e9da3b8-a649-4b34-ada3-35fed1264b09)

### OS

Linux

### GPU

Nvidia

### CPU

AMD

### Ollama version

0.1.47

**Comment by rick-github** [comment]
Seems to work OK with an NVIDIA GeForce RTX 4070, or at least generates sensible output.  In this case, 24 layers get offloaded to the GPU.  If I set layers to 0  (`/set parameter num_gpu 0`) so it all runs on CPU, it also works fine.  But if I set `num_gpu` to 7,as per your log, I get gibberish output.  So number of layers is an influencing factor.

**Comment by Manik04IISER** [comment]
Yes for 0, it's fine but for 7 it's awful and the same is true for 6. If I run it fully on the cpu, performance is bad. So, hopefully someone updates it or perhaps a workaround.

**Comment by Manik04IISER** [comment]
Just an update I have been using the mannix version of the deepseek coder v2 and it's been working perfectly, no crashes, nothing gibberish, and no Chinese. So perhaps please check, what's up with the official one.
